.gitlab-ci.yml

```diff
@@ -53,6 +72,27 @@ sniffers:sonarqube:
   - master
   allow_failure: true
 
+prepare:nodejs:
+  cache:
+    key: "front-packages"
+    paths:
+      - ${THEME_PATH}/node_modules/
+  stage: prepare
+  image: node:lts-alpine
+  before_script:
+  - date
+  - pwd
+  - node -v
+  - yarn -v
+  script:
+  - cd ${THEME_PATH}
+  - yarn install --no-optional --prod
```

Comment:
we have install in first sniffer step then if sniffers fine we use cached node_modules from sniffers and save /dist dir in artifacts to deploy. we don't need node_modules in deploy artifacts

Reply:
It will not work in case of multiple runners, otherwise it will need separate cache service for runner
Moreover push/pull of 260M of node modules to cache and back looks useless comparing to just download fresh packages. 
Also sending to cache means to zip them before send and unzip in new job - x2 more expensive then download or extraction from prebuild image